Doc/library/exceptions.rst

```diff
@@ -965,9 +965,10 @@ their subgroups based on the types of the contained exceptions.
          def derive(self, excs):
             return Errors(excs, self.exit_code)
 
-   Like :exc:`ExceptionGroup`, any subclass of :exc:`BaseExceptionGroup` which
-   is also a subclass of :exc:`Exception` can only wrap instances of
-   :exc:`Exception`.
+   A subclass of :exc:`BaseExceptionGroup` that extends :exc:`Exception`
+   cannot wrap :exc:`BaseException`\ s, as is the case with :exc:`ExceptionGroup`. 
+   However, there is no further restriction: a :exc:`BaseExceptionGroup` subclass
+   which extends :exc:`KeyError` could wrap any :exc:`Exception`.
```

Comment:
I think this is a strange thing to say in a doc, because why would you expect it to not wrap any Exception?

Maybe instead of the second sentence we could just explain the motivation for the restriction: “This ensures that wrapped BaseExceptions are not caught by ‘except Exception’.”

Reply:
> why would you expect it to not wrap any Exception?

Suppose we have `class MyEG(BaseExceptionGroup, X):`, where `except X:` will catch such a group because it extends `X`.  We disallow wrapping `BaseException`s in an `ExceptionGroup` [because they would not be caught by `except Exception:` if unwrapped](https://peps.python.org/pep-0654/#make-it-impossible-to-wrap-baseexceptions-in-an-exception-group).

However this logic is *not* applied consistently: if `X` is e.g. `KeyError`, then `except KeyError:` can indeed catch `MyEG("", [ValueError()])`.  I find this inconsistency surprising, which is why I commented on #99572 and then opened this PR!

(I'm not trying to reopen the design discussion, just trying to explain my perspective)

Reply:
Ok, I see. Well yes, the Exception case is different - people use ‘expect Exception’ as ‘catch everything except fatal errors’.